### PR TITLE
Update HCAL aging parameters and thresholds

### PR DESCRIPTION
**Pull request by kpedro88**
1. Updated HCAL scintillator radiation damage parameters based on latest measurements and fits (https://indico.cern.ch/event/721565/contributions/2966213/attachments/1632432/2603148/2018.04.12_Updated_DvsR_fits_and_predictions.pdf)
2. Updated HCAL SiPM radiation damage parameters based on latest measurements and fits (https://twiki.cern.ch/twiki/pub/CMSPublic/HcalDPGResultsCMSDPS2017042/dark_current_vs_lumi.png)
3. Simplified HCAL PF threshold adjustments for different eras.
4. Use Run 3 PF thresholds for Phase 2 by default (account for increased depth segmentation in HB).
5. Add separate CaloTower thresholds for Run 3 HB depths 1,2 (similar to Phase 1 HE, updated a while back).
6. Update PF and CaloTower thresholds for different HCAL aging scenarios, now that SiPM noise can increase beyond the previous default threshold of ~0.8 GeV.

The new thresholds were proposed, tested, and validated by @hatakeyamak (thanks!): https://baylor.app.box.com/s/w32ja75krcbxcycyifexu28dwlgrj7wg

We observed that PF gammas get a lot of extra energy introduced by ECAL aging; this will be followed up with the appropriate people.

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
Changes in 2019, 2023 workflows expected due to adjustment of the default HB PF thresholds.

Changes in cosmic workflows not expected... will investigate.

**Comment by kpedro88** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by mariadalfonso** [comment]
@slava77 
@kpedro88 PR also solve the issue you opened on the refactoring of the PF thresholds
https://github.com/cms-sw/cmssw/issues/21894

File: CalibCalorimetry/HcalPlugins/python/Hcal_Conditions_forGlobalTag_cff.py [hunk 1]
```diff
@@ -114,7 +114,10 @@
             intlumiOffset = cms.double(150),
             depVsTemp = cms.double(0.0631),
             intlumiToNeutrons = cms.double(3.67e8),
-            depVsNeutrons = cms.vdouble(5.69e-11,7.90e-11),
+            # slopes taken from https://twiki.cern.ch/twiki/pub/CMSPublic/HcalDPGResultsCMSDPS2017042/dark_current_vs_lumi.png
+            # crosstalk divided out
+            # heUpgrade intlumiToNeutrons factor (below) used to convert from /fb-1 to /neutrons 
+            depVsNeutrons = cms.vdouble(5.543e-10,8.012e-10),
```
Comment (perrotta) [inline CalibCalorimetry/HcalPlugins/python/Hcal_Conditions_forGlobalTag_cff.py]: This is a factor 10 larger than the previous values: please confirm that it is correct
(How can I get these numbers from what in https://twiki.cern.ch/twiki/pub/CMSPublic/HcalDPGResultsCMSDPS2017042/dark_current_vs_lumi.png)

Comment (kpedro88) [reply CalibCalorimetry/HcalPlugins/python/Hcal_Conditions_forGlobalTag_cff.py]: Indeed, the prediction has increased by a factor of 10. (This necessitated the re-evaluation of PF thresholds. The previous prediction came from a much older measurement.)

You can get the numbers from the plot by taking the slopes, multiplying by `(1-lambda)` (`lambda` is given as `crosstalk` in the `SiPMCharacteristics` section further down in this file; 0.17 or 0.196 for the different SiPM types), and dividing by the HE dose value `intlumiToNeutrons = cms.double(2.92e7)` (which is in fb-1/neutrons). This is indicated in the comment.

**Comment by pohsun** [comment]
+1

**Comment by perrotta** [comment]
+1
- HCAL aging parameters and CaloTower and PF thresholds are updated according to the most recent fits of data and improved understanding of Phase2
- Reco implementations have been adapted to allow specific thresholds for the different HCAL depths
- It also refactors PF HCAL rechit and cluster configuration, thus fulfilling the request that was in the github issue #21894
- Results were validated and blessed by HCAL
- The updated thresholds lead to an increase of the number of CaloTowers, e.g. jenkins tests output for TTbar14TeV2023D19 wf 20434:
image
and consequently of PF candidates in the same region
image
image
- Higher level physics objects are also affected (Electrons, Photons, Jets, BTag, isolations...) and they must be evaluated by POG/PAG in the release validation
- Modifications are only for 2019 and Phase2 workflows, while there is (by construction) no impact on current year or past data and MC productions

**Pull request merged**